proxy/http/HttpProxyServerMain.cc

```diff
@@ -274,6 +279,55 @@ init_accept_HttpProxyServer(int n_accept_threads)
   }
 }
 
+bool
+et_thread_is_ready()
+{
+  SCOPED_MUTEX_LOCK(lock, startAcceptMutex, this_ethread());
+  if (started_et_net_threads == total_net_threads) {
+    MUTEX_RELEASE(lock);
+    return true;
+  }
+  MUTEX_RELEASE(lock);
+  return false;
+}
+
+AcceptPerThreadCont *accept_per_thread_cont = nullptr;
+BackDoorPerThreadCont *back_door_per_thread_cont = nullptr;
+
+AcceptPerThreadCont::AcceptPerThreadCont() : Continuation(new_ProxyMutex())
+{
+  SET_HANDLER(&AcceptPerThreadCont::handle_event);
+}
+
+BackDoorPerThreadCont::BackDoorPerThreadCont() : Continuation(new_ProxyMutex())
+{
+  SET_HANDLER(&BackDoorPerThreadCont::handle_event);
+}
+
+int
+AcceptPerThreadCont::handle_event(int /* event ATS_UNUSED */, void * /* edata ATS_UNUSED */)
+{
+  if (et_thread_is_ready()) {
```

Comment:
Related to the above, if you use `std::atomic` you can just increment until it's the target value and then go, no locks.

Reply:
yep, fixed